kolibri/tasks/api.py

```diff
@@ -143,68 +169,103 @@ def localdrive(self, request):
+        raise
     connections.close_all()  # close all DB connections (FIX for #1818)
 
-def _localimport(drive_id, update_progress=None):
+def _localimport(drive_id, update_progress=None, check_for_cancel=None):
     drives = get_mounted_drives_with_channel_info()
     drive = drives[drive_id]
-    # copy channel's db file then copy all the content files from sorage dir
-    for channel in drive.metadata["channels"]:
-        call_command(
-            "importchannel",
-            "local",
-            channel["id"],
-            drive.datafolder,
-            update_progress=update_progress)
-        call_command(
-            "importcontent",
-            "local",
-            channel["id"],
-            drive.datafolder,
-            update_progress=update_progress)
+    # copy channel's db file then copy all the content files from storage dir
+    try:
+        for channel in drive.metadata["channels"]:
+            call_command(
+                "importchannel",
+                "local",
+                channel["id"],
+                drive.datafolder,
+                update_progress=update_progress,
+                check_for_cancel=check_for_cancel)
+            call_command(
+                "importcontent",
+                "local",
+                channel["id"],
+                drive.datafolder,
+                update_progress=update_progress,
+                check_for_cancel=check_for_cancel)
+    except UserCancelledError:
+        connections.close_all()  # close all DB connections (FIX for #1818)
+        for channel in drive.metadata["channels"]:
+            channel_id = channel["id"]
+            try:
+                os.remove(get_content_database_file_path(channel_id))
+            except OSError:
+                pass
+            ChannelMetadataCache.objects.filter(id=channel_id).delete()
+        connections.close_all()  # close all DB connections (FIX for #1818)s
+        raise
     connections.close_all()  # close all DB connections (FIX for #1818)
 
 
-def _localexport(drive_id, update_progress=None):
+def _localexport(drive_id, update_progress=None, check_for_cancel=None):
     drives = get_mounted_drives_with_channel_info()
     drive = drives[drive_id]
     for channel in ChannelMetadataCache.objects.all():
         call_command(
             "exportchannel",
             channel.id,
             drive.datafolder,
-            update_progress=update_progress)
-        call_command(
-            "exportcontent",
-            channel.id,
-            drive.datafolder,
-            update_progress=update_progress)
+            update_progress=update_progress,
+            check_for_cancel=check_for_cancel)
+        try:
+            call_command(
+                "exportcontent",
+                channel.id,
+                drive.datafolder,
+                update_progress=update_progress,
+                check_for_cancel=check_for_cancel)
+        except UserCancelledError:
```

Comment:
Oh wait - this is removing a destination for file copy, so the file is never being opened for a connection, as it's not one of the DBs in the content/databases directory, so this is not an issue.